packages/block-library/src/navigation-overlay-close/save.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * WordPress dependencies
+ */
+import { useBlockProps, RichText } from '@wordpress/block-editor';
+import { __ } from '@wordpress/i18n';
+
+export default function NavigationOverlayCloseSave( { attributes } ) {
+	const { displayMode, text } = attributes;
+	const showIcon = displayMode === 'icon' || displayMode === 'both';
+	const showText = displayMode === 'text' || displayMode === 'both';
+
+	// eslint-disable-next-line react-compiler/react-compiler
+	const blockProps = useBlockProps.save( {
+		className: 'wp-block-navigation-overlay-close',
+	} );
+
+	return (
+		<button { ...blockProps } type="button" aria-label={ __( 'Close' ) }>
```

Comment:
We can't use translation functions in the save function, as block validation will fail if the site locale changes. The only way to set localizable default text, as far as I know, is to do it server-side.

Example: https://github.com/WordPress/gutenberg/blob/f46c64a9ab85b5b3d6b8629a952fb66b3496301d/packages/block-library/src/search/index.php#L50

In the first place, I think it might be ideal to dynamically render all content.

- If the icon markup changes in the future, we wouldn't have to worry about block deprecation.
- I don't see any advantage to storing this content in a database.